runtime/include/gpu/chpl-gpu-reduce-util.h

```diff
@@ -43,5 +43,29 @@
   MACRO(chpl_kind, float)   \
   MACRO(chpl_kind, double);
 
+#define GPU_IMPL_SORT_TYPES(MACRO, impl_kind, chpl_kind) \
+  MACRO(impl_kind, chpl_kind, int8_t)  \
+  MACRO(impl_kind, chpl_kind, int16_t)  \
+  MACRO(impl_kind, chpl_kind, int32_t)  \
+  MACRO(impl_kind, chpl_kind, int64_t)  \
+  MACRO(impl_kind, chpl_kind, uint8_t)  \
+  MACRO(impl_kind, chpl_kind, uint16_t)  \
+  MACRO(impl_kind, chpl_kind, uint32_t)  \
+  MACRO(impl_kind, chpl_kind, uint64_t)  \
+  MACRO(impl_kind, chpl_kind, float)   \
+  MACRO(impl_kind, chpl_kind, double);
+
+#define GPU_SORT_TYPES(MACRO, chpl_kind) \
+  MACRO(chpl_kind, int8_t)  \
+  MACRO(chpl_kind, int16_t)  \
+  MACRO(chpl_kind, int32_t)  \
+  MACRO(chpl_kind, int64_t)  \
+  MACRO(chpl_kind, uint8_t)  \
+  MACRO(chpl_kind, uint16_t)  \
+  MACRO(chpl_kind, uint32_t)  \
+  MACRO(chpl_kind, uint64_t)  \
+  MACRO(chpl_kind, float)   \
+  MACRO(chpl_kind, double);
```

Comment:
It looks to me like we could have reused the ones above. Can we do that?

You can use `CUB` here in the macro names as this is way too much detail -- `GPU_CUB_WRAP`, `GPU_DEV_CUB_WRAP` or something?

Reply:
I could reuse the ones above, but it would make it a little harder for someone to follow the code by reading it (it's already pretty hard)

I'm not sure what you mean by too much detail and what you are suggesting I rename. Also I should use CUB in the names here but not in the other code? I guess I'm not following your naming philosophy here.

Reply:
Okay I stared at it and I see what you meant (hopefully):
- Rename file to `chpl-gpu-cub-util.h`
- Rename `GPU_IMPL_REDUCE` to `GPU_DEV_CUB_WRAP` and `GPU_REDUCE` to `GPU_CUB_WRAP`
- Remove `GPU_IMPL_SORT_TYPES` and `GPU_SORT_TYPES`

Reply:
Yeah, I like those changes. (Again, sorry for quick/terse comments).